FilterDataGrid.Net/FilterDataGrid.cs

```diff
@@ -1807,6 +1814,11 @@ private void PopupPlacement(FrameworkElement grid, FrameworkElement header)
             }
         }
 
+        private void ItemSource_CollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            ItemsSourceCount = Items.Count;
+            OnPropertyChanged(nameof(ItemsSourceCount));
```

Comment:
You need to renumber the rows here, sometimes deleting or adding rows from the middle of a collection will cause numbering issues for the row headers.

`            // Renumber all rows
            for (int i = 0; i < this.Items.Count; i++)
            {
                var row = this.ItemContainerGenerator.ContainerFromIndex(i) as DataGridRow;
                if (row != null)
                {
                    row.Header = (i + 1).ToString();
                }
            }`

Reply:
My tests have shown that this is not necessary or is already implemented elsewhere.
The issue I faced was that the Status bar was not updating with the ItemsSourceCount when an ObservableCollection was modified,
The code changes simply listen for ObservableCollection.CollectionChanged event and updates the property.

Reply:
Can you confirm you tried removing items from the middle of an observablecollection and have the numbering shown to be updated in the correct order? cause I tested this yesterday and it didnt work without this code.

Reply:
My apologies, the tests I ran where using the filters and not modifying the ObservableCollection.
I can confirm that previously removing elements from the collection would cause the row numberings to contain gaps.
I have applied the suggested code.
Good spot 👍